Tasks/Inc/BPS_Fifo.h

```diff
@@ -0,0 +1,174 @@
+/**
+ *  Copyright Longhorn Racing Solar 2022-2023 
+ *  Implements a FIFO structure on Micrium's Heap
+ * 
+*/
+#ifndef __BPS_FIFO_H
+#define __BPS_FIFO_H
+
+// Includes
+#include "os.h"
+#include "lib_mem.h"
+#include <stdint.h>
+#include <stdbool.h>
+
+// Flags for the Fifo
+typedef enum {
+    FIFO_INITIALIZED,
+    FIFO_EMPTY,
+    FIFO_FULL,
+    FIFO_AVAILABLE,
+    FIFO_ERR
+} fifo_flags;
+
+// Fifo struct
+typedef struct {
+    size_t fifoSize;
+    size_t elementSize;
+    intptr_t* putptr;
+    intptr_t* getptr;
+    intptr_t* fifo; // range of fifo base address -> fifo base address[fifoSize - 1]
+    unsigned int flags;
+} BPS_Fifo_t;
+
+// An invalid fifo - returned on errors
+const BPS_Fifo_t INVALID_FIFO = {0, 0, NULL, NULL, NULL, FIFO_ERR};
+
+/**
+ * @brief Returns size of an element in the FIFO
+ * 
+ * @param fifo 
+ * @return size_t 
+ */
+inline size_t fifoTypeSize(BPS_Fifo_t* fifo) {
```

Comment:
Probably don't need this PR anymore, but I would use static inline to prevent compiler optimization.